libs/partners/cohere/langchain_cohere/chat_models.py

```diff
@@ -60,19 +58,17 @@ def get_cohere_chat_request(
     Returns:
         The request for the Cohere chat API.
     """
-    documents = (
-        None
-        if "source_documents" not in kwargs
-        else [
-            {
-                "snippet": doc.page_content,
-                "id": doc.metadata.get("id") or f"doc-{str(i)}",
-            }
-            for i, doc in enumerate(kwargs["source_documents"])
-        ]
-    )
-    kwargs.pop("source_documents", None)
-    maybe_connectors = connectors if documents is None else None
+    additional_kwargs = messages[-1].additional_kwargs
+
+    # cohere SDK will fail loudly if both connectors and documents are provided
+    documents = [
+        {
+            "snippet": doc.page_content,
+            "id": doc.metadata.get("id") or f"doc-{str(i)}",
+        }
+        for i, doc in enumerate(additional_kwargs.get("documents", []) or [])
+    ]
+    connectors = [{"id": connector} for connector in additional_kwargs.get("connectors", []) or []]
```

Comment:
`None` is now filtered out from being sent to the API
```suggestion
    connectors = [{"id": connector} for connector in additional_kwargs.get("connectors", []) or None]
```

Reply:
actually, sorry, the `if not connectors: connectors = None` is a slightly better approach than the suggestion I made

Reply:
👍